raiden/raiden_event_handler.py

```diff
@@ -127,6 +127,58 @@ def unlock(
     )
 
 
+def _update_lock_info(
+    state_to_update: NettingChannelState,
+    update_from_state: NettingChannelState,
+    chain_state: ChainState,
+) -> None:
+    """Update old channel state with lock information from current state
+
+    Call this after settlement, when you need to work on an older locksroot
+    because the channel has been settled with an outdated balance proof.
+    """
+    assert update_from_state.settle_transaction, "Channel must be settled"
+    for old_state, new_state in [
+        (state_to_update.our_state, update_from_state.our_state),
+        (state_to_update.partner_state, update_from_state.partner_state),
+    ]:
+        # After settlement, no unlocks can be processed. So all locks that are
+        # still present in the balance proof are locked, unless their secret is
+        # registered on-chain.
+        old_state.secrethashes_to_lockedlocks.update(
+            {
+                secret: unlock.lock
+                for secret, unlock in old_state.secrethashes_to_unlockedlocks.items()
+            }
+        )
+        old_state.secrethashes_to_unlockedlocks = {}
+
+        # In the time between the states, some locks might have been unlocked
+        # on-chain. Update their state to "on-chain unlocked".
+        for secret, updated_unlock in new_state.secrethashes_to_onchain_unlockedlocks.items():
+            try:
+                del old_state.secrethashes_to_lockedlocks[secret]
+            except KeyError:
+                continue
+            old_state.secrethashes_to_onchain_unlockedlocks[secret] = updated_unlock
+
+        # If we don't have a task for the secret, then that lock can't be
+        # relevant to us, anymore. Otherwise, we would not have deleted the
+        # payment task.
+        # One case where this is necessary: We are a mediator and didn't unlock
+        # the receiver's BP, but the secret has been registered on-chain. We
```

Comment:
You're right. I'm still not really happy with this paragraph, because we're talking about a payer and a payee, which makes me assume that those are payer and payee the same transfer at first thought. But they aren't:

```
Initiator ----> Mediator ----> Target
       payer payee    payer payee
       ^^^^^                ^^^^^
```

But we do this in many other places (e.g. MediationPairState) in the same way, so it's consistent and thus acceptable.